### PR TITLE
Drop py37

### PR DESCRIPTION
**Pull request by daico007**
To coordinate with foyer, this PR drop support for py37. Also fix a bug coming from a arg name change in foyer (i.e., this PR should only pass the bleeding test and none of the other). This PR can wait until release foyer get released.

### CODE REVIEW

**Review by CalCraven — APPROVED** [review]
Looks like there are still some failing tests in test_impropers_parameterization.py due to the symbol vs element changes done here to align with Foyer. So either we need to wait until a foyer release to merge this, or pin the testing to the develop version of Foyer.

**Comment by daico007** [comment]
Bleeding test passed and everything else (as expected), so I am gonna merge this PR

**Pull request merged**
